src/coreComponents/physicsSolvers/fluidFlow/ThermalCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -482,53 +514,46 @@ class FaceBasedAssemblyKernel : public isothermalCompositionalMultiphaseFVMKerne
                                      stack.thermalTransmissibility,
                                      stack.dTrans_dPres ); // again, we have to pass something here, but this is unused for now
 
-    // Step 2: compute temperature difference at the interface
-    for( integer i = 0; i < stack.stencilSize; ++i )
-    {
-      localIndex const er  = m_seri( iconn, i );
-      localIndex const esr = m_sesri( iconn, i );
-      localIndex const ei  = m_sei( iconn, i );
 
-      stack.energyFlux += stack.thermalTransmissibility[0][i] * m_temp[er][esr][ei];
-      stack.dEnergyFlux_dT[i] += stack.thermalTransmissibility[0][i];
-    }
 
-    // **********************************************************************************
-    // At this point, we have computed the energyFlux and the compFlux for all components
-    // We have to do two things here:
-    // 1) Add dCompFlux_dTemp to the localFluxJacobian of the component mass balance equations
-    // 2) Add energyFlux and its derivatives to the localFlux(Jacobian) of the energy balance equation
+    // Step 2: compute temperature difference at the interface
+    localIndex k[2];
+    localIndex connectionIndex = 0;
 
-    // Step 1: add dCompFlux_dTemp to localFluxJacobian
-    for( integer ic = 0; ic < numComp; ++ic )
+    for( k[0] = 0; k[0] < stack.numConnectedElems; ++k[0] )
     {
-      for( integer ke = 0; ke < stack.stencilSize; ++ke )
+      for( k[1] = k[0] + 1; k[1] < stack.numConnectedElems; ++k[1] )
       {
```

Comment:
Since you do not loop over `k`, maybe a simpler solution like 
```c++
    for( int k0 = 0; k0 < stack.numConnectedElems; ++k0 )
    {
      for( int k1 = k0 + 1; k1 < stack.numConnectedElems; ++k1 )
      {
```
This reduces the visibility of `k`.
And also, is it the same `2` as in `numFluxSupportPoints `?
Here, it's a `2` because it's the size of a pair, isn't it? Would it change with other schemes?
We could also even rename `k0, k1` with other names?

Reply:
This assembly pattern will only work for a two point scheme. Honestly, I would not spend too much time refining this. I feel this looping pattern should be generalized and moved somewhere else so that it can be used both here and in the singlePhase kernels. Right now they look identical so I would prefer keeping this just as it is and then change both at once.

Reply:
OK